GoogleContainerTools/skaffold
# Fix tarring of build context for artifacts with source dependencies

Fixes: #5749  

**Description**

Main change is in:
 -  `dependencies.go`: split method `ResolveForArtifact` into methods `SingleArtifactDependencies` and `TransitiveArtifactDependencies`
 - `cloud_build.go`: for `cloud build` the tar context should only include source dependencies from the target artifact.
 - `run_test.go`: add integration test for running `examples/microservices` with `googleCloudBuild`.

All other changes are from interface and method renaming.

**Testing instructions**
In `integration/examples/microservices` directory skaffold build should succeed with google cloud builder:
```
skaffold build -p gcb --default-repo=<image_repo>
```
